From b1f73b0b68a4c82a27c65a935afa48b8b9933e1e Mon Sep 17 00:00:00 2001
From: "Ethan D. Twardy" <ethan.twardy@gmail.com>
Date: Sat, 10 May 2025 21:01:18 -0500
Subject: [PATCH 1/2] dovecot: Include dovecot-config in dovecot-dev

This tool is used by dovecot plugins (e.g. pigeonhole) to locate and
build against dovecot libraries.

diff --git a/meta-networking/recipes-support/dovecot/dovecot_2.3.21.1.bb b/meta-networking/recipes-support/dovecot/dovecot_2.3.21.1.bb
index 3177cdb1d7..d1e8fee878 100644
--- a/meta-networking/recipes-support/dovecot/dovecot_2.3.21.1.bb
+++ b/meta-networking/recipes-support/dovecot/dovecot_2.3.21.1.bb
@@ -49,7 +49,10 @@ SYSTEMD_SERVICE:${PN} = "dovecot.service dovecot.socket"
 SYSTEMD_AUTO_ENABLE = "disable"
 
 do_install:append () {
-    rm -rf ${D}${libdir}/dovecot/dovecot-config
+    sed -e "s/-f\(debug\|macro\)-prefix-map=[^[:space:]]*//g" \
+        -e "s@-I${RECIPE_SYSROOT}@-I@g" \
+        -i ${D}${libdir}/dovecot/dovecot-config
+
     install -d 755 ${D}/etc/dovecot
     touch ${D}/etc/dovecot/dovecot.conf
     chmod 644 ${D}/etc/dovecot/dovecot.conf
@@ -61,6 +64,14 @@ do_install:append () {
     oe_multilib_header dovecot/config.h
 }
 
+SYSROOT_PREPROCESS_FUNCS += "dovecot_sysroot_preprocess"
+dovecot_sysroot_preprocess () {
+    install -Dm755 ${D}${libdir}/dovecot/dovecot-config -t ${SYSROOT_DESTDIR}${libdir}/dovecot
+    sed -e "s@^LIBDOVECOT='-L${libdir}@LIBDOVECOT='-L${RECIPE_SYSROOT}${libdir}@" \
+        -e "s@-I${includedir}@-I${RECIPE_SYSROOT}${includedir}@" \
+        -i ${SYSROOT_DESTDIR}${libdir}/dovecot/dovecot-config
+}
+
 USERADD_PACKAGES = "${PN}"
 USERADD_PARAM:${PN} = "-r -d ${libexecdir} -M -s ${base_sbindir}/nologin -g dovecot dovecot; \
                       -r -d ${libexecdir} -M -s ${base_sbindir}/nologin -g dovenull dovenull"
@@ -70,7 +81,8 @@ FILES:${PN} += "${libdir}/dovecot/*plugin.so \
                 ${libdir}/dovecot/libfs_compress.so \
                 ${libdir}/dovecot/libssl_iostream_openssl.so"
 FILES:${PN}-staticdev += "${libdir}/dovecot/*/*.a"
-FILES:${PN}-dev += "${libdir}/dovecot/libdovecot*.so"
+FILES:${PN}-dev += "${libdir}/dovecot/libdovecot*.so \
+                ${libdir}/dovecot/dovecot-config"
 FILES:${PN}-dbg += "${libdir}/dovecot/*/.debug"
 
 CVE_STATUS[CVE-2016-4983] = "not-applicable-platform: Affects only postinstall script on specific distribution."
-- 
2.47.2

